applications/FluidDynamicsApplication/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_explicit_cpp_template_with_integration.cpp

```diff
@@ -0,0 +1,603 @@
+    const double &DN_DX_0_0 = data.DN_DX(0, 0);
+    const double &DN_DX_0_1 = data.DN_DX(0, 1);
+    const double &DN_DX_1_0 = data.DN_DX(1, 0);
+    const double &DN_DX_1_1 = data.DN_DX(1, 1);
+    const double &DN_DX_2_0 = data.DN_DX(2, 0);
+    const double &DN_DX_2_1 = data.DN_DX(2, 1);
+
+    //substitute_rhs_2D
+
+    // Here we assume that all the weights of the gauss points are the same so we multiply at the end by Volume/n_nodes
+    rRightHandSideVector *= data.volume / static_cast<double>(n_nodes);
+
+    KRATOS_CATCH("")
+}
+
+template<>
+void CompressibleNavierStokesExplicit<3>::CalculateRightHandSide(
+    VectorType &rRightHandSideVector,
+    const ProcessInfo &rCurrentProcessInfo)
+{
+    KRATOS_TRY
+
+    constexpr unsigned int n_nodes = 4;
+    constexpr unsigned int block_size = 5;
+    constexpr unsigned int matrix_size = n_nodes * block_size;
+
+    if (rRightHandSideVector.size() != matrix_size) {
+        rRightHandSideVector.resize(matrix_size, false); //false says not to preserve existing storage!!
+    }
+
+    // Struct to pass around the data
+    ElementDataStruct data;
+    this->FillElementData(data, rCurrentProcessInfo);
+
+    // Substitute the formulation symbols by the data structure values
+    const double h = data.h;
+    const array_1d<double, n_nodes> &r = data.r;
+    const BoundedMatrix<double, n_nodes, 3> &f_ext = data.f_ext;
+    const double mu = data.mu;
+    const double lambda = data.lambda;
+    const double c_v = data.c_v;
+    const double gamma = data.gamma;
+    const double v_sc = data.nu_sc;
+    const double k_sc = data.lambda_sc;
+
+    // Stabilization parameters
+    const double stab_c1 = 4.0;
+    const double stab_c2 = 2.0;
+
+    // Solution vector values from nodal data
+    // This is intentionally done in this way to limit the matrix acceses
+    // The notation U_i_j DOF j value in node i
+    const double &U_0_0 = data.U(0, 0);
+    const double &U_0_1 = data.U(0, 1);
+    const double &U_0_2 = data.U(0, 2);
+    const double &U_0_3 = data.U(0, 3);
+    const double &U_0_4 = data.U(0, 4);
+    const double &U_1_0 = data.U(1, 0);
+    const double &U_1_1 = data.U(1, 1);
+    const double &U_1_2 = data.U(1, 2);
+    const double &U_1_3 = data.U(1, 3);
+    const double &U_1_4 = data.U(1, 4);
+    const double &U_2_0 = data.U(2, 0);
+    const double &U_2_1 = data.U(2, 1);
+    const double &U_2_2 = data.U(2, 2);
+    const double &U_2_3 = data.U(2, 3);
+    const double &U_2_4 = data.U(2, 4);
+    const double &U_3_0 = data.U(3, 0);
+    const double &U_3_1 = data.U(3, 1);
+    const double &U_3_2 = data.U(3, 2);
+    const double &U_3_3 = data.U(3, 3);
+    const double &U_3_4 = data.U(3, 4);
+
+    // Hardcoded shape functions for linear tetrahedra element
+    // This is explicitly done to minimize the alocation and matrix acceses
+    // The notation N_i_j means shape function for node j in Gauss pt. i
+    const double N_0_0 = 0.58541020;

```

Comment:
same as above. in a future PR this should go in the symbolic gen

Reply:
Done.